### ISSUE TITLE
DEP

### ISSUE THREAD

**Issue by arnesmits** [Bioconductor/Contributions#439]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/arnesmits/DEP

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Subversion][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/source-control/
[6]: http://bioconductor.org/developers/how-to/git-mirrors/

**Comment by mtmorgan** [comment]
DESCRIPTION

- Consider wrapping long lines to 80 columns

- Carefully consider the long list of `Imports:` as each dependency increases the susceptibility of your package to other changes.

- Consider addtional biocViews terms

vignette

- line 42: Is `setInternet2()` needed?

- 109: adopt consistent use of pipes, e.g., no `.` as arguments here.

- The vignette looks really good, thanks!

R

- `stop(paste(.))` constructs (e.g., functions.R:29) can often be replaced by `stop(.)`.

- consider more stringent assertions, e.g., for length and non-NA values.

- For arguments such as `meode` (functions.R:224) consider specifying alternatives in the function signature and using `match.arg()` for validation.

- create local variables, e.g., `idx <- is.na(assay(se))` to avoid repeated calls; also consider the optimization `assay(., withDimnames = FALSE)` to suppress expensive (for large assays) addition of dim names.

- functions.R:346 `rowsum()` on `t(assay(.))` may be a more straight-forward way of calculating per-group values

- functions.R:441 and elsewhere: use `seq_len()` or `seq_along()` instead of `1:n` to avoid problems when `n == 0`. `rnorm()` can take vectors of mean and sd, so probably this iteration can be replacced by a single vectorized call.

- functions.R:533 typo: "run impute() with..."

- functions.R:643: maybe as a 'too cute' expression, consider revising to `apply(combn(condtions, 2), 2, paste, collapse = " - ")`

- functions.R:936: use consistent indenting; the formatR or lintr package might be helpful.

- `run_app_function.R`: assert that `app` is a scalar character, else the `if` statement will fail in a cryptic way.

- I believe that 'best practice' in the tidyverse, in package code, is to use the `_` (e.g., `select_()` rather than `select()`) form of verbs, and hence avoid the need non-standard evaluation and for `utils::globalVariables()`. I am not completely sure of this.

man

- Generally good. Consider using language to describe parameters using R terminology and specifying length and other constraints, e.g., add_rejections.Rd:14 `\item{alpha}{numeric(1), sets the ...`.

inst/shiny_apps

- e.g., inst/shiny_apps/TMT/:210: be sure to avoid too much code in the app, which is not subject to R CMD check; instead, write functions accessible (maybe not exported) in the R/ directory. Ideally, the app.R file will contain the minimum code to connect server-side functionality to the implementation, without any implementation itself. 

- pay attention to code formatting in these files, especially avoiding long therefore hard to parse lines

Other

- inst/NEWS: be careful to keep this accurate and in a format that can be parsed by `utils::news()`.

- nice to see the relatively extesive tests!

**Comment by arnesmits** [comment]
Dear Martin,

Thank you for your review! I have carefully read your comments, edited appropriate parts of the package and pushed the changes. The edits include the following points:

Description
1)	Wrapped longer lines in the description.
2)	Carefully checked the Imports list and could only reduce it minimally. The package utilizes many functionalities from other packages, requiring them to be imported.
3)	Additional biocViews (“DataRepresentation”) added.

Vignette
4)	Implemented consistent use of pipes, thank you for noticing! 
5)	setInternet2() indeed does not seem to be necessary, so line 42 is deleted.

R
6)	The paste() functions are removed from the stop() constructs.
7)	More stringent assertions are implemented, checking for argument lengths in most functions and non-NA values in some functions.
8)	Arguments with fixed alternatives are now indicated in the function signature and are checked with match.arg(). Thanks for pointing out this neat solution!
9)	Implemented novel local variables to avoid repeated calls, especially those for accessing rowData and colData.
10)	Thank you for noting the combination of rowsum() and t(assay(.)). Eventually, I did not implement it, because the current construct allows easy filtering based on dplyr.
11)	Seq_len() is implemented in the appropriate parts. 
12)	functions.R:484 - originally functions.R:441 Although rnorm can take vectors of mean and sd, it cannot take a vector of the numbers of observations (if a vector is applied, it will take its length as the number of observations). Because we need different numbers of observations for the different samples, it looks like this cannot be simplified in a single vectorized call.
13)	Typo is corrected. 
14)	Implemented paste() directly into the apply function. 
15)	The tidyverse indeed used to work with underscore suffixed verbs, but these are currently deprecated (dplyr >= 7.0.0). 

man
16)	Manuals have been edited to inform users about the class and length of parameters.

inst/shiny_apps
17)	Appropriate parts of the shiny apps are reformatted (reducing long lines) and several internal (i.e. not exported) functions are implemented. This leaves the apps with only minimal implementation. 

Other
18)	New ‘NEWS.Rd’ file added, which can be appropriately parsed.
I am looking forward to your response.

Kind regards,
Arne

**Comment by mtmorgan** [comment]
Thanks for the changes and response; this looks great and I am accepting the package. We will work through adding it to the Bioconductor git repository; you should be updated in a day or so.

**Comment by arnesmits** [comment]
Great. Thank you for your efforts!

**Comment by mtmorgan** [comment]
The master branch of your GitHub repository has been added to
Bioconductor's git repository. See instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/
  
to keep your GitHub and Bioconductor repositories in sync.

Your package should be included in the next nigthly 'devel' build
(check-out at about 6 pm Eastern; build completion around 2pm
Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main
landing page are consistent with the addition of your
package). Once the package builds successfully, you package will be
available for download in the 'Devel' version of Bioconductor using
`biocLite("YOUR_PACKAGE_NAME")`. The package 'landing page' will
be created at

https://bioconductor.org/packages/YOUR_PACKAGE_NAME

If you have any questions, please contact the bioc-devel mailing
list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue
will not be monitored further.
